geekhours/database.py

```diff
@@ -79,6 +80,9 @@ def insert_donelist(self, date: str, course: str, duration: str):
         Insert donelist into the 'donelist' table.
         The course name must be a registered name in the 'course' table.
         """
+        if date is not str:
```

Comment:
BTW, Is the possible type of date is not str?

Reply:
Fixed it to use `isinstance()`.
Cast the `date` to `str`, this is because when the `geekhours done add` command is executed without the `-d` or `--date` option, today's date of `datetime.date` type is passed to `date`,

```
# bin/geekours

def initialize():
    ...
    today = datetime.strptime(str(date.today()), '%Y-%m-%d').date()
    ...

def get_args(today: str):
    ...
    done_add_parser.add_argument(
        '-d',
        '--date',
        nargs='?',
        default=today,
    ...
```
and the TypeError occurs  when validating YYYY-MM-DD format using len().
```
$ geekhours done add eng
Traceback (most recent call last):
  File "/home/yukie/.local/bin/geekhours", line 299, in <module>
    main()
  File "/home/yukie/.local/bin/geekhours", line 292, in main
    args.handler(args, db_path)
  File "/home/yukie/.local/bin/geekhours", line 123, in add_done
    cmd.insert_donelist(args.date, args.course_name, args.duration)
  File "/home/yukie/.local/lib/python3.6/site-packages/geekhours/command.py", line 35, in insert_donelist
    self.database.insert_donelist(date, course, duration)
  File "/home/yukie/.local/lib/python3.6/site-packages/geekhours/database.py", line 101, in insert_donelist
    if not len(date) == 10:
TypeError: object of type 'datetime.date' has no len()
```
However, since annotated to pass `str` in 
```
# bin/geekours

def get_args(today: str):
...
```
and
```
# database.py

def insert_donelist(self, date: str, course: str, duration: str):
...
```
it should be cast in the `def initialize()` of geekhours, not in database.py.

Reply:
Fixed it to cast when initializing geekhours.
```
def initialize():
    ...
    today = datetime.strptime(str(date.today()), '%Y-%m-%d').date()
    today = str(today)
```